ports/winit/app.rs

```diff
@@ -62,6 +71,83 @@ impl App {
             windows: HashMap::new(),
         };
 
+        struct Minibrowser {
+        context: egui_glow::EguiGlow,
+        location: RefCell<String>,
+        }
+
+        impl Minibrowser {
+            fn update(&mut self, window: &winit::window::Window) {
+                let Self { context, location } = self;
+                let _duration = context.run(window, |ctx| {
+                    TopBottomPanel::top("toolbar").show(ctx, |ui| {
+                        ui.allocate_ui_with_layout(
+                            ui.available_size(),
+                            egui::Layout::right_to_left(egui::Align::Center),
+                            |ui| {
+                                if ui.button("go").clicked() {
+                                    // TODO go
+                                    dbg!("go clicked");
+                                }
+                                ui.add_sized(
+                                    ui.available_size(),
+                                    egui::TextEdit::singleline(&mut *location.borrow_mut()),
+                                );
+                            },
+                        );
+                    });
+
+                    unsafe {
+                        TOOLBAR_HEIGHT = ctx.used_rect().height();
+                    }
+                });
+                context.paint(window);
+            }
+        }
+
+        // Create FrameBufferObject
+        let webrender_surfman = window.webrender_surfman();
+        let webrender_gl = match webrender_surfman.connection().gl_api() {
+            GLApi::GL => unsafe { gl::GlFns::load_with(|s| webrender_surfman.get_proc_address(s)) },
+            GLApi::GLES => unsafe {
+                gl::GlesFns::load_with(|s| webrender_surfman.get_proc_address(s))
+            },
+        };
+
+        // Make sure the gl context is made current.
+        webrender_surfman.make_gl_context_current().unwrap();
+        debug_assert_eq!(webrender_gl.get_error(), gleam::gl::NO_ERROR,);
+
+        // Bind the webrender framebuffer
+        let framebuffer_object = webrender_surfman
+            .context_surface_info()
+            .unwrap_or(None)
+            .and_then(|info| NonZeroU32::new(info.framebuffer_object))
+            .map(NativeFramebuffer);
+
+        let gl = unsafe {
+            glow::Context::from_loader_function(|s| {
+                webrender_surfman.get_proc_address(s)
+            })
+        };
+
+        // glow needs to set framebuffer as a target
+        unsafe {
+            gl.bind_framebuffer(glow::FRAMEBUFFER, framebuffer_object);
+        };
```

Comment:
when i run this now, the fbo number is zero, which apparently tells [glBindFramebuffer](https://registry.khronos.org/OpenGL-Refpages/gl4/html/glBindFramebuffer.xhtml) to break any existing binding. i wonder if we really need this?

Reply:
removed in e9f8c3c0bbbdc6166f9006dc69d88a11103e2b2d